src/temperature.rs

```diff
@@ -245,6 +278,83 @@ mod test {
         assert_almost_eq(o, 180.0);
     }
 
+    #[test]
+    #[cfg(feature = "from_str")]
+    fn empty_str() {
+        let t = Temperature::from_str("");
+        assert!(t.is_ok());
+
+        let o = t.unwrap().as_celsius();
+        assert_eq!(o, 0.0);
+    }
+
+    #[test]
+    #[cfg(feature = "from_str")]
+    fn celsius_str() {
+        let t = Temperature::from_str("100C");
```

Comment:
What about `100.0 deg C`, or `100.0 °C`?

Reply:
Added solution for both of these cases and tests, please review!